### DISCUSSION TITLE
Import "solcx" could not be resolved

### DISCUSSION THREAD

**Discussion by yarg119** [smartcontractkit/full-blockchain-solidity-course-py#1184] [Q&A] [answered]
I keep getting this error. anyone know what I can do about this. I just got to the point where I am using Visual Studio Code, and this error seems to one that can jack up an entire contract. I downloaded all of the dependencies and troubleshooting to get rid of all of the errors that were coming up during installation. I've been working on this for about two hours now, and I need some help.

`solcx.install_solc: The term 'solcx.install_solc' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.`

*Update* when I try to deploy

**Comment by yarg119** [comment]
I did not fix this issue, however, I was able to run deploy.py

**Comment by cromewar** [comment, marked as answer]
Hello @yarg-eth did you change the python interpreter on vscode (bottom left corner) and choose the one you have solc-x installed on?

**Reply by yarg119** [reply]
No, I had not. This fixed the error message. Thank you so much.

**Reply by yarg119** [reply]
To clarify, if you have some extra extensions installed. It is the curly brackets on the bottom bar of your vscode that allows one to change their python interpreter.

**Reply by marescialloNino** [reply]
I have a similar problem, @cromewar , i installed solcx, however it keeps giving me this error, i tried changing the interpreter but it doesn't work!
solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed. Use solcx.install_solc('0.6.0') to install.

**Reply by cromewar** [reply]
Hello @nixon-crypto please refer to [chronological-issues-from-video.md](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md) there you will have more info about some lines you must add to that lesson in order to work.

**Reply by marescialloNino** [reply]
thank you @cromewar

**Comment by JohnEMagee** [comment]
As far as I can tell the plug in is 'broken' with higher versions of Python.  If I can figure out a virtual environment with an earlier version (3.8 maybe) it might work.  There's no indication that anyone is working on a fix for this so a note should be added to the video regarding what version of Pythons it works with

**Comment by solanamagic** [comment]
Can anyone assist? I keep getting the following error:

\demos\web3_py_simplestorage\deploy.py", line 1, in <module>
    from solcx import compile_standard, install_solc
ModuleNotFoundError: No module named 'solcx'

Had this resolved and then it came back, cant continue the course for the time being.
Here is my code:

from solcx import compile_standard, install_solc
import json

from web3 import Web3

install_solc("0.6.0")

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)

bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]

w3 = Web3(Web3.HTTPProvider("HTTP://127.0.0.1:7545"))
chain_id = 5777
my_address = "0x3748F7ACa1d2cadeb0b52F6E72CEA058E1760526"
private_key = "0x73dc41d1cd1bcb782bd9300d3eae36d1c475c59a9ee463c77012ac11039ef903"

SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)
print(SimpleStorage)

**Reply by yarg119** [reply]
You probaly installed solcx "0.8.12" and you are trying to call "0.6.0" in your code. Try switching it to "0.8.12" because that is what your computer installs now when you run the command to install it.

**Reply by solanamagic** [reply]
Gave it a try but I just can seem to get it to work, the same error keeps getting thrown up even when I change the version info

**Reply by JohnEMagee** [reply]
There's a line of code that installs the right version, it's unlikely a version issue.  The plug in is not working properly with newer versions of python...not sure it's still an active project working on fixing it but the plugin website itself says it's 'faililng'.

**Reply by yarg119** [reply]
That's weird because it works for me on  Python 3.10.12. The fix for me was uninstalling all of my old Python versions and installing my Python 310 in C:\ Then reinstalling solcx in cmd prompt.

**Reply by JohnEMagee** [reply]
I'm working on a mac so not sure if that makes a difference, i did install solcx using pip3 - i do all my python work in python3 (and was using sublime text for most of it until now)

**Reply by yarg119** [reply]
It honestly probably does make a difference. I have had a lot of trouble on Windows that some of you mac users don't have to deal with in this course, but I'm sure you guys experience your fair share of OS based problems as well.

**Reply by cromewar** [reply]
@JohnEMagee there is actually a difference between python and python3 on mac (as python uses python 2.7) are you using a m1 chip mac?

**Reply by JohnEMagee** [reply]
I am yes - in the thread I started myself regarding this issue I spelled it out, I'm using an m1 mac mini - i know it has issues with certain things but I usually find work arounds, didn't find one for this yet...

**Reply by solanamagic** [reply]
I will give it another try today but if the plug-in is broken does that mean that I will struggle to continue with the course?

**Reply by JohnEMagee** [reply]
Maybe?  I've made it through to 6 just by switching to solana 0.8.12 - you have to do some tweaking along the way but it's not  huge.  I'm waiting on an answer about the chainlink plug ins because that causes an issue too since you can't use solidity 0.6.0 to compile those either

**Reply by yarg119** [reply]
@solanamagic you might have to reinstall everything. Once I got to the part with brownie, I had to uninstall python and everything with that. What I would do is start fresh, uninstall everything that you can and reinstall it but make sure to delete the leftover directories after uninstallation. Some of this is probably completely unrelated with your problem now, but it will save you a lot of time and similar problems later in the course.
